Physical_Engine/Query/BottomCentreline.cs

```diff
@@ -0,0 +1,65 @@
+﻿using BH.Engine.Geometry;
+using BH.Engine.Reflection;
+using BH.Engine.Spatial;
+using BH.oM.Base;
+using BH.oM.Geometry;
+using BH.oM.Physical.FramingProperties;
+using BH.oM.Reflection.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Configuration;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+
+        [Description("Returns the bottom centreline of an IFramingElement")]
+        [Input("BHoM.Physical.Elements.IFramingElement", "The IFramingEment to query the bottom centreline of")]
+        [Output("ICurve", "The bottom centreline of the IFramingElement")]
+
+        public static ICurve BottomCentreline(this BH.oM.Physical.Elements.IFramingElement element)
+        {
+            ICurve location = element.Location;
+
+            Vector normal;
+            
+            if(location.IIsLinear())
+            { 
+                normal = BH.Engine.Physical.Query.Normal(element);
+            }
+            else
+            {
+                Engine.Reflection.Compute.RecordError("IFramingElement must have linear location line");
```

Comment:
Could be worth specifying a bit more what method is failing. This method could be called from somewhere else, and then having an error message that refers more to were things went wrong can be helpful.
```suggestion
                Engine.Reflection.Compute.RecordError("Can only extract bottom centrelines from linear IFramingElements.");
```

Reply:
How many centrelines can a framing element have? Surely it should be `Can only extra bottom centreline from linear IFramingElements.` - singular `centreline`?